### PR TITLE
Eip micropayments standard for nfts and multi tokens

### PR DESCRIPTION
**Pull request by julesl23**
Hi,

Please find my new submission for an EIP. My email address is _jlai@fabstir.com_ if you wish to contact me via email.

Kind regards,
Jules Lai

### CODE REVIEW

**Comment by julesl23** [comment]
@poojaranjan 
Thanks, I added the FEM link to the proposal.

**Comment by poojaranjan** [comment]
@julesl23 
Thanks for making the changes. 
It will be reviewed by EIP editors and will be merged with the main repo if approved.   

@alita-moore 
I don't see any editor being pinged in this pull request. Can you check the bot for any possible bug?

**Comment by alita-moore** [comment]
@poojaranjan this is intended behavior when a file is not named correctly. What behavior would you prefer? Also please open a issue in eth-bot 😊

**Comment by julesl23** [comment]
> @poojaranjan this is intended behavior when a file is not named correctly. What behavior would you prefer? Also please open a issue in eth-bot 😊

@alita-moore 
Opened an issue for the EIP file name problem. Not sure if I opened it correctly though

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:2]:
```suggestion
eip: 4393
```

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:5]:
```suggestion
author: Jules Lai (@julesl23)
```

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:11]:
```suggestion
requires: 165, 721, 1155
```

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:16]: Please use relative links of the form `[EIP-X](./eip-X.md)`

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:18]:
```suggestion
This standard proposal outlines a generalised way to allow tipping via implementation of an `ITipToken` interface. The interface is intentionally kept to a minimum in order to allow for maximum use cases.
```

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:42]:
```suggestion
A cheap way to send tips to any type of NFT or multi token. This can be achieved by gas optimising the tip token contract and sending the tips in batches using the `tipBatch` function from the interface.
```
Please wrap code text in backticks.

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:237]:
```suggestion
Tip token values are exchanged with ERC-20 deposits and vice-versa. It is left to the tip token implementer to decide on the price of a tip token and hence how much tip to mint in exchange for the ERC-20 deposited.
```

Comment (lightclient) [reply EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:237]: Please update all occurrences of unhyphenated ERC/EIPs.

File: assets/eip-####/README.md
Comment (lightclient) [inline assets/eip-####/README.md:1]: Please remove this file.

Comment (lightclient) [inline assets/eip-####/README.md:1]: Please remove this file.

Comment (lightclient) [inline assets/eip-####/README.md:1]: Please remove this file.

**Review by lightclient — CHANGES_REQUESTED** [review]

**Comment by lightclient** [comment]
FYI I think it would be better to just include the `*.sol` files rather than entire project structure. We wouldn't want to merge the `package.lock` for instance.

**Comment by alita-moore** [comment]
> FYI I think it would be better to just include the `*.sol` files rather than entire project structure. We wouldn't want to merge the `package.lock` for instance.

is this intended for me? Can you add this to the issue? https://github.com/ethereum/EIP-Bot/issues/5 I will work on it this weekend.

**Comment by MicahZoltu** [comment]
> > FYI I think it would be better to just include the `*.sol` files rather than entire project structure. We wouldn't want to merge the `package.lock` for instance.
> 
> is this intended for me? Can you add this to the issue? [ethereum/EIP-Bot#5](https://github.com/ethereum/EIP-Bot/issues/5) I will work on it this weekend.

No, it is to the author of this EIP.

**Comment by julesl23** [comment]
> > > FYI I think it would be better to just include the `*.sol` files rather than entire project structure. We wouldn't want to merge the `package.lock` for instance.
> > 
> > 
> > is this intended for me? Can you add this to the issue? [ethereum/EIP-Bot#5](https://github.com/ethereum/EIP-Bot/issues/5) I will work on it this weekend.
> 
> No, it is to the author of this EIP.

Yep, and I made those changes.

**Comment by alita-moore** [comment]
Makes sense. That's embarrassing 😅

File: assets/eip-####/contracts/TipToken.sol
Comment (axic) [inline assets/eip-####/contracts/TipToken.sol:21]: This is an implementation. Standards should not provide implementations, but interfaces.

Comment (julesl23) [reply assets/eip-####/contracts/TipToken.sol]: > This is an implementation. Standards should not provide implementations, but interfaces.

I removed the implementation and added in 'Reference Implementation' section of proposal a link to my public repo for example implementation and tests

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (MicahZoltu) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:309]:
```suggestion
```
External links should not be included in EIPs.  Either include an inline reference implementation, or files in `../assets/eip-4393/x.sol`.  Alternatively, this is an optional section and can simply be left out if that makes the most sense.

Comment (julesl23) [reply EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md]: > External links should not be included in EIPs. Either include an inline reference implementation, or files in `../assets/eip-4393/x.sol`. Alternatively, this is an optional section and can simply be left out if that makes the most sense.

All sorted in the latest commit. Let me know if there is anything else I need to do.

**Comment by julesl23** [comment]
Hi, issue #4398 has been closed. Am I now waiting on editor approval for merge to move the PR to draft phase? Or is there anything else for me to do at this stage?

**Comment by alita-moore** [comment]
@julesl23 you'll need to assign the eip an eip number which will be provided by the editors

**Comment by alita-moore** [comment]
> A critical exception has occurred (cc @alita-moore):
> 	Message: not found
> 

😑
I'm not near my computer for the next few days so I'll address this then.

**Comment by Andrel2s** [comment]
Jump

**Comment by Andrel2s** [comment]
Hi

**Comment by MicahZoltu** [comment]
Blocked by CI:
```
./EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:105: everytime ==> every time
./EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:218: idential ==> identical
```
Once that is resolved you'll need to wait for an editor to approve it (lightclient or axic most likely).

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:14]: The abstract should be a short, technical description of the EIP. Please update this before moving forward into review.

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:50]:
```suggestion
```
Implied via EIP-1

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (lightclient) [inline EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:52]: Need to replace EIP-xxxx -- usually prefer to not refer to the EIP itself by its own number. Instead say "this EIP" or a something similar.

**Review by lightclient — CHANGES_REQUESTED** [review]

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (julesl23) [reply EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md:14]: Thanks. Updated.

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (julesl23) [reply EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md]: Done

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (julesl23) [reply EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md]: Done

**Comment by julesl23** [comment]
Hi, please let me know if there is anything else before approval/review. Thanks

File: EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md
Comment (MicahZoltu) [reply EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md]: Alternatively you can do `Julian Bushell (@jules123)`.  It is the author's choice what name they wish to go by.

**Review by lightclient — APPROVED** [review]

**Comment by alita-moore** [comment]
"pull request not in correct state to enable automerge" from enable-auto-merge

**Pull request merged**
